### PR TITLE
Added support for Algos and application call

### PR DESCRIPTION
**Pull request by rmeena840**
Added support for Algos and application call
Task: https://www.pivotaltracker.com/story/show/181767866

### CODE REVIEW

File: wallet/src/App.vue [hunk 1]
```diff
@@ -10,36 +10,41 @@
...
-import { WalletType } from "./types/enum.types";
-import { ChainType } from "./constants";
+import { WalletType } from "./types";
+import { toAddress, CHAIN_NAME } from "./constants";
 import WalletStore from "./store/WalletStore";
 import {
   MyAlgoWalletSession,
   WallectConnectSession,
   WebMode,
+  types,
```
Comment (vuvoth) [inline wallet/src/App.vue]: I think we should rename it to wtypes

Comment (vuvth) [inline wallet/src/App.vue:38]: I think we should rename it to wtypes

File: wallet/src/App.vue [hunk 2]
```diff
@@ -129,7 +133,49 @@ export default defineComponent({
     handleLogOut() {
       console.log("Wallet Disconnected");
       this.walletAddress = "";
-      this.setWalletType(WalletType.NONE);
+      this.walletStore.setWalletType(WalletType.NONE);
+    },
+    async executeTx(amount: number) {
```
Comment (vuvoth) [inline wallet/src/App.vue:151]:
```suggestion
    async executeTx(amount: bigint | number) {
```

Comment (vuvth) [inline wallet/src/App.vue:138]:
```suggestion
    async executeTx(amount: bigint | number) {
```

File: wallet/src/App.vue [hunk 3]
```diff
@@ -129,7 +133,49 @@ export default defineComponent({
...
+            sign: types.SignType.SecretKey,
+            fromAccount: { addr: this.walletAddress, sk: new Uint8Array(0) },
+            toAccountAddr: toAddress,
+            amountMicroAlgos: amount,
+            payFlags: { totalFee: 1000 },
+          },
+        ];
+        let response = await webMode.executeTx(txParams);
+        this.transactionMessage =
+          amount / 1e6 +
```
Comment (vuvoth) [inline wallet/src/App.vue]: What is 1e6 meaning in this case? Let add notation about it.

Comment (vuvth) [inline wallet/src/App.vue:153]: What is 1e6 meaning in this case? Let add notation about it.

File: wallet/src/App.vue [hunk 4]
```diff
@@ -129,7 +133,49 @@ export default defineComponent({
...
+    },
+    async executeAppTx() {
+      try {
+        const webMode = this.walletStore.getWebMode;
+        const execParams: types.ExecParams[] = [
+          {
+            type: types.TransactionType.CallApp,
+            sign: types.SignType.SecretKey,
+            fromAccount: { addr: this.walletAddress, sk: new Uint8Array(0) },
+            appID: 189,
```
Comment (vuvoth) [inline wallet/src/App.vue]: Why appID is constant here?

Comment (vuvth) [inline wallet/src/App.vue:169]: Why appID is constant here?

File: wallet/src/App.vue [hunk 5]
```diff
@@ -129,7 +133,49 @@ export default defineComponent({
...
+            payFlags: { totalFee: 1000 },
+          },
+        ];
+        let response = await webMode.executeTx(execParams);
+        this.transactionMessage = "App has been deployed.";
```
Comment (vuvoth) [inline wallet/src/App.vue]: We don't deploy new app here. Let change the messages

Comment (vuvth) [inline wallet/src/App.vue:174]: We don't deploy new app here. Let change the messages

File: wallet/src/constants/index.ts [hunk 6]
```diff
@@ -3,3 +3,8 @@ export enum ChainType {
     MainNet = "MainNet",
     TestNet = "TestNet",
```
Comment (vuvoth) [inline wallet/src/constants/index.ts]: What if we want to use local(private) network? Example is sandbox.

Comment (vuvth) [inline wallet/src/constants/index.ts:4]: What if we want to use local(private) network? Example is sandbox.

**Review by vuvoth — COMMENTED** [review]
Thanks! I left few comments.

File: wallet/src/App.vue (hunk 1 above)
Comment (rmeena840) [reply wallet/src/App.vue]: Changed to wtpes.

File: wallet/src/App.vue (hunk 4 above)
Comment (rmeena840) [reply wallet/src/App.vue]: applicationID has been added as parameter similar to amount in above method for sending algos.

File: wallet/src/App.vue (hunk 5 above)
Comment (rmeena840) [reply wallet/src/App.vue]: ok. Changed to `App has been called.`

File: wallet/src/constants/index.ts (hunk 6 above)
Comment (rmeena840) [reply wallet/src/constants/index.ts]: another property `Private` has been added in enum to accommodate this feature.

File: wallet/src/App.vue (hunk 3 above)
Comment (rmeena840) [reply wallet/src/App.vue]: utility method has been added for conversion to Algo unit.

File: wallet/src/types/enum.types.ts [hunk 7]
```diff
@@ -4,3 +4,10 @@ export enum WalletType {
 	MY_ALGO = "My Algo Connect",
 	WALLET_CONNECT = "Wallet Connect",
 }
+
+// algosigner ledger name [ MainNet | TestNet | <private_net_display_name> ]
+export enum ChainType {
+	MainNet = "MainNet",
+	TestNet = "TestNet",
+	Private = "your_private_netowkr"  // repliace it with your private network
```
Comment (rmeena840) [inline wallet/src/types/enum.types.ts]:
```suggestion
	Private = "your_private_netowkr"  // replace it with your private network
```

File: wallet/src/App.vue [hunk 8]
```diff
@@ -10,36 +10,42 @@
...
-    >
-      Disconnect {{ selectedWallet }}
-    </button>
+    <div class="header" v-if="walletAddress">
+      <p>Address: {{ walletAddress }}</p>
+      <button type="button" class="walletButton" @click="executeTx(10e6)">
+        Send 10 Algo
+      </button>
+      <button type="button" class="walletButton" @click="executeAppTx(189)">
+        Application Call
```
Comment (Megha-Dev-19) [inline wallet/src/App.vue]: Can you move this application ID to constants so that it can be easily configured by the user

Comment (rmeena840) [reply wallet/src/App.vue]: ok. Moved to constant fil.

**Review by vuvoth — APPROVED** [review]

**Pull request merged**